src/server/assets/assetsFilter/index.js

```diff
@@ -0,0 +1,23 @@
+const assetRegex = type =>
+  new RegExp(`(\\/static\\/js\\/(${type})-\\w+\\.\\w+\\.js)`, 'g');
+
+const assetsFilter = (assets, service) => {
+  const serviceAssets = assets.filter(asset =>
+    asset.match(assetRegex(service)),
+  );
+  const vendorAssets = assets.filter(asset =>
+    asset.match(assetRegex('vendor')),
+  );
+  const mainAssets = assets.filter(asset => asset.match(assetRegex('main')));
+
+  /*
+   * Service bundles must appear first in the page so react-loadable
+   * knows to ensure theyre preloaded before hydration is done.
+   */
+  const orderedAssets = [...serviceAssets, ...vendorAssets, ...mainAssets];
+
+  // Filter out duplicates just incase
```

Comment:
Do we have an example of getting duplicates? Are duplicate script tags that worrying, surely the browser would only request the actual asset it once?

Reply:
One of the issues that test had was the filter was causing assets to appear twice in script tags, which caused a react error.